workflowproj/resources.go

```diff
@@ -21,38 +21,45 @@ import (
 	libopenapiutils "github.com/pb33f/libopenapi/utils"
 	"github.com/santhosh-tekuri/jsonschema/v5"
 	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type ResourceKind int
 
-	"github.com/kiegroup/kogito-serverless-operator/api/metadata"
+const (
+	OpenApiResource ResourceKind = iota
+	AsyncApiResource
+	CamelRouteResource
+	GenericResource
 )
 
-// ParseResourceType tries to parse the contents of the given resource and find the correct type.
+// ParseResourceKind tries to parse the contents of the given resource and find the correct type.
 // Async and OpenAPI files are pretty fast to parse (0.00s).
 // Camel and generic files can take a fair price from the CPU (0.03s on the i5) since it takes more processing power.
-func ParseResourceType(contents string) metadata.ExtResType {
+func ParseResourceKind(contents []byte) ResourceKind {
```

Comment:
Why change the signature, the content from the configmap is returned as a string and only here https://github.com/kiegroup/kogito-serverless-operator/pull/150/files#diff-b07f106da84e1629f797884f0abd0c38ccca6628da08911e656c19e9da9edec5R42 is asked a byte[]

Reply:
This is not used anymore, I just kept it here because can be useful in the future to guess the types and make a more useful/lightweight manifest for users. `[]byte` is usually returned from marshallers/conversions/read functions, that's why I've change it.